tests/legacy-cli/e2e_runner.ts

```diff
@@ -215,10 +216,19 @@ process.env.CHROME_BIN = path.resolve(process.env.CHROME_BIN!);
 process.env.CHROME_PATH = path.resolve(process.env.CHROME_PATH!);
 process.env.CHROMEDRIVER_BIN = path.resolve(process.env.CHROMEDRIVER_BIN!);
 
-Promise.all([findFreePort(), findFreePort(), findPackageTars()])
-  .then(async ([httpPort, httpsPort, packageTars]) => {
-    setGlobalVariable('package-registry', 'http://localhost:' + httpPort);
-    setGlobalVariable('package-secure-registry', 'http://localhost:' + httpsPort);
+// Find free ports sequentially, reducing the risk of collisions.
+// Note for Windows test mode on CI: Verdaccio runs inside WSL, so we
+// never want to reserve a port in the Windows host.
+const ports = (async () => {
+  const portA = await findFreePort({ neverRunOnWslHost: true });
+  const portB = await findFreePort({ neverRunOnWslHost: true });
+  return [portA, portB];
+})();
+
+Promise.all([ports, findPackageTars()])
+  .then(async ([[httpPort, httpsPort], packageTars]) => {
+    setGlobalVariable('package-registry', 'http://127.0.0.1:' + httpPort);
+    setGlobalVariable('package-secure-registry', 'http://127.0.0.1:' + httpsPort);
```

Comment:
If this is the `httpsPort` should the URL schema also be `https`?

Reply:
Apparently "secure" means "authentication required" 😄 so this is left as it was before